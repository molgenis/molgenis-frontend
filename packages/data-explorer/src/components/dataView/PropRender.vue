<template>
  <div>
    <div v-for="(value, propertyName, index) in object" :key="index">
      <template v-if="(typeof value !== 'object')">
        {{ propertyName }}: {{ value }}
      </template>
      <template v-else-if="Array.isArray(value)">
        {{ propertyName }}: {{ value.map(v => v.label).join(', ') }}
      </template>
      <template v-else>
        {{ propertyName }}:
        <div v-for="(refValue, refPropertyName, refIndex) in value" :key="refIndex" class="pl-3">
          {{ refPropertyName }}: {{ refValue }}
        </div>
      </template>
    </div>
  </div>
</template>

<script>
export default {
  name: 'PropRender',
  props: {
    object: {
      type: Object,
      required: true
    }
  }
}
</script>
