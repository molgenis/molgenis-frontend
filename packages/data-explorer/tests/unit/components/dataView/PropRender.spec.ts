import { shallowMount } from '@vue/test-utils'
import PropRender from '@/components/dataView/PropRender.vue'

describe('PropRender.vue', () => {
  const propsData = {
    object: {
      'col1': 'Column 1',
      'col2': ['Column 2', 'Column 2b'],
      'col3': {
        'col4': 'Col 4'
      }
    }
  }
  const wrapper = shallowMount(PropRender, { propsData })
  it('exists', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
