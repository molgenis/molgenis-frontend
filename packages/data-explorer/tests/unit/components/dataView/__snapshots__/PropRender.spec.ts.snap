// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropRender.vue exists 1`] = `
<div>
  <div>
    col1: Column 1
  </div>
  <div>
    col2: ,
  </div>
  <div>
    col3:
    <div class="pl-3">
      col4: Col 4
    </div>
  </div>
</div>
`;
